test/gurka/test_warnings.cc

```diff
@@ -0,0 +1,239 @@
+#include "gurka.h"
+#include "test.h"
+
+using namespace valhalla;
+
+std::string deprecated_costing_methods[3] = {"auto_shorter", "hov", "auto_data_fix"};
+
+// test case for routes endpoint
+TEST(warnings, routes_endpoint) {
+  const std::string ascii_map = R"(
+          A----------B-------------C-----P
+          |          |                   |
+          |          |                   |
+          |          3--------U     G----S
+          |          |        |     |    |
+          D----------E--------4-----F----L
+     )";
```

Comment:
I think we can reduce the complexity drastically here ;) it's good that you're testing all the things you added. this is what I'd change:
- have a single map for everything and an easy one. we're only testing deprecated params (so far) so the map doesn't really matter (yet). a `A--B--C` is totally fine. otherwise people (me) will be stunned in the future and frantically look for the reason for such a detailed map ;)
- look into other gurka files, you'll see how you can use gtest's `SetUpTestSuite` function that only runs once and all test fixtures can use the same map
- (bonus point, but not really necessary): parameterized test suite with just one function calling all permutations of endpoints and deprecated parameters. wasn't sure if it's possible with gtest, but seems so: https://stackoverflow.com/a/59377529

Reply:
:smile: thank you for this. having one map for all the test cases really simplified it. The file even looks way cleaner.